readthedocs/projects/models.py

```diff
@@ -659,6 +660,17 @@ def proxied_static_path(self):
         """Path for static files hosted on the user's doc domain."""
         return f"{self.proxied_api_host}/static/"
 
+    @property
+    def custom_path_prefix(self):
+        """
+        Get the path prefix from the custom urlconf.
+
+        Returns `None` if the project doesn't have a custom urlconf.
+        """
+        if self.urlconf:
+            return self.urlconf.split("$", 1)[0]
```

Comment:
Still don't love this logic, but don't have an obviously better suggestion. I think it could still use a comment explaining the thought:

```suggestion
            # Return the value before the first defined variable,
            # as that is a prefix and not part of our normal doc patterns. 
            return self.urlconf.split("$", 1)[0]
```